### ISSUE TITLE
Video recording limit on HTC M8

### ISSUE THREAD

**Issue by resetcoder** [Divested-Mobile/DivestOS-Build#203]
I've just installed DivestOS on my HTC One M8, because I need to record uninterrupted videos for 8-10 hours, and as far as I know the 4GB file size limit is removed only on Android 11+ versions. The installed OS is working fine, I have a 64GB SD card in the phone formatted to Exfat filesystem, so there is no file size limitation by the FS.

As I've noticed the default camera application is OpenCamera, which is a kind of nice application, so I've checked all its settings, there is not video length or video size limit set, and restart recording at file size limit is set to enabled.

I've just started a test recording, and one hour later when I checked the phone it was not recording at all.

Is there still a 4GB file size limit in DivestOS?
Is it possible the OpenCamera app couldn't write a larger file than 4GB?
If it has reached the file size limit and stopped because of this reason, why didn't it restart recording as it was configured?

I couldn't check the recorded video yet because I've restarted the recording with a 2GB file size limit to check if it continues and the recording are seamless, but it behaves now like before with Android 6.0, so this installation seems completely pointless for me for now.

If there is no OS level file size limit then what camera application should I use that can record more than 30 mins/4GB?

Thanks.

**Comment by SkewedZeppelin** [comment]
>  I checked the phone it was not recording at all.

Did the recording simply stop? Or did the app crash? Or did the system crash?

> record uninterrupted videos for 8-10 hours,

I have strong doubts this is viable.

**Comment by resetcoder** [comment]
> Did the recording simply stop? Or did the app crash? Or did the system crash?
It simply stopped, I could continue recording just buy touching the record button, the app/system did not crash.

> I have strong doubts this is viable.
I've just did that with a Huawei phone a few weeks ago, I've recorded from 11pm to 8am, I've got a single mp4 file with size ~50GB, it had Android 11, full stock, using the built in camera app.

[XdaDevelopers: Android 11 may finally remove Android's 4GB file size limit for video recordings](https://www.xda-developers.com/android-11-unlimited-video-recording-file-size/)
[AndroidPolice: Android 11 got rid of the 4GB limit on videos, but the Google Camera app is still capped](https://www.androidpolice.com/2020/09/26/android-11-got-rid-of-the-4gb-limit-on-videos-but-the-google-camera-app-is-still-capped/)

The previous articles describe the situation, OpenCamera should handle files over 4GB, but I don't know the version implemented in DivestedOS, it may be an older app that may not support the new API?

What should I do to find out why it can not record more?

Btw I've just checked, the phone was still recording with 2GB size limit set to OpenCamera, with auto restart enabled. But I'm not sure yet if the slicings of the videos are seamless, and I need like 4 devices recording the same event, it will be a nightmare to merge and synchronize all recordings if they are sliced to smaller videos.

**Comment by resetcoder** [comment]
**Update:** I've copied everything from the phone to my PC and the largest video was indeed 4GB. I couldn't even copy files with size larger than 4GB from the PC to the phone. I've checked the SD card again just to be sure and it was formatted to exFat (I've formatted it in my laptop under Windows).

Then I formatted the SD card again but now inside the phone, using DivestedOS built in formatter, and then the built-in OpenCamera actually **_could_** create videos larger than 4GB. I've stopped the test at 7GB, it works. It seems this android doesn't like the windows-formatted filesystem, even though it didn't have file size limitation at 4GB.

Issue solved.
